nist/int.go

```diff
@@ -314,21 +313,17 @@ func (i *Int) UnmarshalBinary(buf []byte) error {
 	if len(buf) != i.MarshalSize() {
 		return errors.New("Int.Decode: wrong size buffer")
 	}
-	i.V.SetBytes(buf)
+	if i.endianness == binary.LittleEndian {
+		i.SetLittleEndian(buf)
+	} else {
+		i.V.SetBytes(buf)
```

Comment:
SetBytes should be sensitive to the current byte-order, so this if-else shouldn't be needed here; it should happen in SetBytes instead.